### PR TITLE
[feat]: [CDS-41440]: Add ARM/BP endpoints from CG to NG

### PR DESCRIPTION
**Pull request by nassergonzalez**
[CDS-41440]: Add ARM/BP endpoints from CG to NG

## Describe your changes
For the implementation of Azure ARM BP in next gen, we need 2 new endpoints to retrieve the management groups given a subscription and the location of the deployment zones
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

[CDS-41440]: https://harness.atlassian.net/browse/CDS-41440?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 14 Issues 
* Code issue 12 Code issues 
* Anti-pattern 2 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/36385)

**Review by abosii-harness — CHANGES_REQUESTED** [review]
> Reviewed 7 of 11 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: 4 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/resources/azure/AzureResource.java` line 323 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XIFpPDRc1OlA-XqCI:-N9XIFpQ5UrH5QBEd-cm:b-kz76kk) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/120-ng-manager/src/main/java/io/harness/ng/core/resources/azure/AzureResource.java#L323)):*
> ```Java
> 
>   @GET
>   @Path("locations-by-subscription")
> ```

this path should be just locations, logic can be if (isNotEmpty(subscriptionId)), I don't see why we would require a new endpoint here
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureLocationsDTO.java` line 25 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XIlP76dt599MtMlhS:-N9XIlP76dt599MtMlhT:bsajc98) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureLocationsDTO.java#L25)):*
> ```Java
> @Value
> @Builder
> @FieldDefaults(level = AccessLevel.PRIVATE)
> ```

This is already by default PRIVATE for @Value
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureLocationsDTO.java` line 29 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XIeSPA4yAeaWUB2jF:-N9XIeSPA4yAeaWUB2jG:b-6afu8l) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureLocationsDTO.java#L29)):*
> ```Java
> @OwnedBy(CDP)
> @Schema(name = "AzureLocationsDTO", description = "Azure response for locations")
> ```

this space is not required
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureManagementGroupsDTO.java` line 25 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XIpKS0btYEFPr4bYF:-N9XIpKS0btYEFPr4bYG:bsajc98) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureManagementGroupsDTO.java#L25)):*
> ```Java
> @Value
> @Builder
> @FieldDefaults(level = AccessLevel.PRIVATE)
> ```

This is already by default PRIVATE for @Value
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java` line 250 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XJ6Tc1N0r6VmD7uxe:-N9XJ6Tc1N0r6VmD7uxf:bbwtu7f) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java#L250)):*
> ```Java
>     BaseNGAccess baseNGAccess =
>         azureHelperService.getBaseNGAccess(connectorRef.getAccountIdentifier(), orgIdentifier, projectIdentifier);
>     List<EncryptedDataDetail> encryptionDetails = azureHelperService.getEncryptionDetails(connector, baseNGAccess);
> ```

we need to check this, after recent changes  this call may allow access to unathorized resources. Can you check with @vlica-harness what can be done here

**Review by abosii-harness — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: 4 file(s) to review. 1 change requests. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java` line 250 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XJ6Tc1N0r6VmD7uxe:-N9XJns59EJgNxI_5woM:b-4wihx4) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java#L250)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

we need to check this, after recent changes  this call may allow access to unathorized resources. Can you check with @vlica-harness what can be done here
</blockquote></details>

nvm, this is not merged yet.. my bad..

**Review by tmacari-harness — CHANGES_REQUESTED** [review]
> Reviewed 4 of 11 files at r1, 2 of 6 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: 4 file(s) to review. 2 change requests. 0 of 1 approvals obtained. 

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceService.java` line 41 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b1EOzCzhShgYDKyRC:-N9b1EP-7YXDZny_pVJL:bruftu4) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceService.java#L41)):*
> ```Java
>       IdentifierRef connectorRef, String orgIdentifier, String projectIdentifier, String subscriptionId);
> 
>   AzureManagementGroupsDTO getResourceManagement(
> ```

maybe getAzureManagementGroups ?
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java` line 265 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b1bgJ0JZbL4A3XeaA:-N9b1bgJ0JZbL4A3XeaB:bd0l7ax) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java#L265)):*
> ```Java
>     List<EncryptedDataDetail> encryptionDetails = azureHelperService.getEncryptionDetails(connector, baseNGAccess);
>     Map<AzureAdditionalParams, String> additionalParams = new HashMap<>(Collections.emptyMap());
>     if (!Objects.equals(subscriptionId, "")) {
> ```

can be better: if (isNotEmpty(subscriptionId))
___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureAsyncTaskHelper.java` line 705 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b2fIs94hDmeHOBo_-:-N9b2fIs94hDmeHOBo_0:b-j0lf8j) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureAsyncTaskHelper.java#L705)):*
> ```Java
>             .locations(Arrays.stream(Region.values())
>                            .filter(region
>                                -> (AzureEnvironmentType.AZURE_US_GOVERNMENT == azureEnvironmentType)
> ```

A bit confusing, whats the condition here?
___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureTask.java` line 66 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b3Z6z3R6s1bEqqMDN:-N9b3Z6z3R6s1bEqqMDO:b-xutz6l) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureTask.java#L66)):*
> ```Java
>             azureTaskParams.getEncryptionDetails(), azureTaskParams.getAzureConnector());
>       case LIST_SUBSCRIPTION_LOCATIONS:
>         if (azureTaskParams.getAdditionalParams().size() == 0) {
> ```

I think it would be more safe to check for specific property we then for size.

**Review by nassergonzalez — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: 7 file(s) to review. 2 change requests. 8. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/resources/azure/AzureResource.java` line 323 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XIFpPDRc1OlA-XqCI:-N9XS8fN3kv00WU1HEGb:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/120-ng-manager/src/main/java/io/harness/ng/core/resources/azure/AzureResource.java#L323)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

this path should be just locations, logic can be if (isNotEmpty(subscriptionId)), I don't see why we would require a new endpoint here
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureLocationsDTO.java` line 25 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XIlP76dt599MtMlhS:-N9XSI252sph8RU7w_Zr:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureLocationsDTO.java#L25)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

This is already by default PRIVATE for @Value
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureLocationsDTO.java` line 29 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XIeSPA4yAeaWUB2jF:-N9XSPwbBQkFivxn76h-:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureLocationsDTO.java#L29)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

this space is not required
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureManagementGroupsDTO.java` line 25 at r1](https://reviewable.io/reviews/harness/harness-core/36385#-N9XIpKS0btYEFPr4bYF:-N9XSV2E1b3zhq8O--3e:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/5252e53dcc9be4f97c1e23902ba7913d24b3545e/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/dtos/AzureManagementGroupsDTO.java#L25)):*
<details><summary><i>Previously, abosii-harness (Alexandru Bosii) wrote…</i></summary><blockquote>

This is already by default PRIVATE for @Value
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceService.java` line 41 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b1EOzCzhShgYDKyRC:-N9b9BgmBB3hTyC_j2H_:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceService.java#L41)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

maybe getAzureManagementGroups ?
</blockquote></details>

Done.
___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java` line 265 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b1bgJ0JZbL4A3XeaA:-N9bAouTAu8__cT4yA_Z:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java#L265)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

can be better: if (isNotEmpty(subscriptionId))
</blockquote></details>

Done.
___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureAsyncTaskHelper.java` line 705 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b2fIs94hDmeHOBo_-:-N9bBaOE4M44nDciVjdF:bs40e5a) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureAsyncTaskHelper.java#L705)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

A bit confusing, whats the condition here?
</blockquote></details>

Filters out the Azure_US_GOVERMENT regions unless the environment is an AZURE_US_GOVERNMENT environment type
___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureTask.java` line 66 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b3Z6z3R6s1bEqqMDN:-N9bCKMO7UdNjm3HGOhp:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureTask.java#L66)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

I think it would be more safe to check for specific property we then for size.
</blockquote></details>

Done.

**Comment by nassergonzalez** [comment]
trigger compile

**Review by tmacari-harness — CHANGES_REQUESTED** [review]
> Reviewed 5 of 6 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: 1 file(s) to review. 2 change requests. 4. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureAsyncTaskHelper.java` line 705 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b2fIs94hDmeHOBo_-:-N9ewO2A4YWMEKhqBJBH:bxje1p3) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureAsyncTaskHelper.java#L705)):*
<details><summary><i>Previously, nassergonzalez (Nasser Gonzalez Gonzalez) wrote…</i></summary><blockquote>

Filters out the Azure_US_GOVERMENT regions unless the environment is an AZURE_US_GOVERNMENT environment type
</blockquote></details>

I mean the full condition for filtering it's confusing for me: 
```(AzureEnvironmentType.AZURE_US_GOVERNMENT == azureEnvironmentType) == AzureUtils.AZURE_GOV_REGIONS_NAMES.contains(region.name())```

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 1 of 11 files at r1.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: 1 change requests. 4. unresolved discussions.

File: 120-ng-manager/src/main/java/io/harness/ng/core/resources/azure/AzureResource.java [hunk 1]
```diff
@@ -302,4 +304,33 @@ private InfrastructureDefinitionConfig getInfrastructureDefinitionConfig(
...
+    return ResponseDTO.newResponse(
+        azureResourceService.getAzureManagementGroups(connectorRef, orgIdentifier, projectIdentifier));
+  }
+
+  @GET
+  @Path("locations")
+  @ApiOperation(value = "Gets azure locations defined for a subscription", nickname = "getLocationsBySubscription")
+  public ResponseDTO<AzureLocationsDTO> getLocations(
+      @NotNull @QueryParam("connectorRef") String azureConnectorIdentifier,
+      @QueryParam("subscriptionId") String subscriptionId,
```
Comment (vlica-harness) [inline 120-ng-manager/src/main/java/io/harness/ng/core/resources/azure/AzureResource.java:327]: Could we have @NotNull @NotEmpty for subscriptionId ?

**Review by abosii-harness — DISMISSED** [review]
> Reviewed 1 of 11 files at r1, 2 of 6 files at r2, 5 of 6 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: :shipit: complete! ✓

___
*[`125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java` line 258 at r5](https://reviewable.io/reviews/harness/harness-core/36385#-N9fHn2R78daTZnNhzjq:-N9fHn2S5CbvrSRH-hUs:bqpc0mc) ([raw file](https://github.com/harness/harness-core/blob/ba67cd4ee3aad6f68357afbdbe36cda023c2f4a9/125-cd-nextgen/src/main/java/io/harness/cdng/k8s/resources/azure/service/AzureResourceServiceImpl.java#L258)):*
> ```Java
> 
>   @Override
>   public AzureLocationsDTO getLocationsBySubscription(
> ```

nit: name can be just getLocations since subscription id here is optional

**Review by nassergonzalez — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: Some of the checks are too old. 2. unresolved discussions. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/resources/azure/AzureResource.java` line 327 at r5](https://reviewable.io/reviews/harness/harness-core/36385#-N9XBf_AQBzb966A6ILF-r5-327:-NA3bVWR9Ak_APAGHuY4:b-rx08lf) ([raw file](https://github.com/harness/harness-core/blob/ba67cd4ee3aad6f68357afbdbe36cda023c2f4a9/120-ng-manager/src/main/java/io/harness/ng/core/resources/azure/AzureResource.java#L327)):*
<details><summary><i>Previously, vlica-harness (Vasile Lica) wrote…</i></summary><blockquote>

Could we have @NotNull @NotEmpty for subscriptionId ?
</blockquote></details>

Subscription is optional. If subscription is not present, we will return all the regions, and if the subscription is present we will return the regions allowed for that subscription
___
*[`930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureAsyncTaskHelper.java` line 705 at r3](https://reviewable.io/reviews/harness/harness-core/36385#-N9b2fIs94hDmeHOBo_-:-NA3dYAx8XRaKCt4OgxX:b-7ui6zn) ([raw file](https://github.com/harness/harness-core/blob/aa96dfa6adb420469f88f674b659e488a066b7ab/930-delegate-tasks/src/main/java/software/wings/delegatetasks/azure/AzureAsyncTaskHelper.java#L705)):*
<details><summary><i>Previously, tmacari-harness (Tudor Macari) wrote…</i></summary><blockquote>

I mean the full condition for filtering it's confusing for me: 
```(AzureEnvironmentType.AZURE_US_GOVERNMENT == azureEnvironmentType) == AzureUtils.AZURE_GOV_REGIONS_NAMES.contains(region.name())```


</blockquote></details>

Maybe we can work together here but for me it's clear:
Filter the regions where if this is an environmentType where we use AzureGoverment regions, get only the region where the name of the region matches the region

**Review by vlica-harness — DISMISSED** [review]
> Reviewed 4 of 11 files at r1, 2 of 6 files at r2, 1 of 6 files at r4, 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: :shipit: complete! ✓

**Review by puthrayaharness — DISMISSED** [review]
> Reviewed 4 of 11 files at r1, 2 of 6 files at r2, 1 of 6 files at r4, 5 of 5 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: :shipit: complete! ✓

**Review by tmacari-harness — APPROVED** [review]
> Reviewed 4 of 5 files at r6, 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: :shipit: complete! ✓

**Comment by nassergonzalez** [comment]
trigger codebasehash

**Review by puthrayaharness — APPROVED** [review]
> Reviewed 1 of 1 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/36385)* status: :shipit: complete! ✓

**Pull request merged**